cmake/SetupCompilerOptions.cmake

```diff
@@ -111,6 +111,17 @@ if(BLT_DEFINES)
     message(STATUS "Added \"${BLT_DEFINES}\" to definitions")
 endif()
 
+if(COMPILER_FAMILY_IS_MSVC)
+    # Mark 32 bit executables large address aware so they can use > 2GB address space
+    if(CMAKE_SIZEOF_VOID_P MATCHES 4)
+        SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /LARGEADDRESSAWARE")
+    endif()
+
+    # Visual studio can give a warning that /bigobj is required due to the size of some object files
+    set( BLT_CXX_FLAGS "${BLT_CXX_FLAGS} /bigobj" )
+    set( BLT_C_FLAGS   "${BLT_C_FLAGS} /bigobj" )
+endif()
```

Comment:
This one seems less perilous:

https://msdn.microsoft.com/en-us/library/ms173499.aspx